cmd/skopeo/main.go

```diff
@@ -99,8 +103,22 @@ func main() {
 	if reexec.Init() {
 		return
 	}
-	rootCmd, _ := createApp()
-	if err := rootCmd.Execute(); err != nil {
+	rootCmd, opts := createApp()
+	retry(rootCmd, opts)
+}
+
+func retry(rootCmd *cobra.Command, opts *globalOptions) {
+	err := rootCmd.Execute()
```

Comment:
It’s not clear that `.Execute` is safe to call multiple times; at the very least this will cause the warning in `opts.before` to be printed on every call.